peps/pep-0784.rst

```diff
@@ -0,0 +1,300 @@
+modules will be updated to discuss the planned deprecation and removal
+timelines.
+
+
+Backwards Compatibility
+=======================
+
+The main compatibility concern is usage of existing standard library
+compression APIs with the existing import names. These names will be
+deprecated in 3.19 and will be removed in 3.24. Given the long coexistance of
+the modules and a 5 year deprecation period, most users will likely migrate to
+the new import names well before then. Additionally, a libCST codemod can be
+provided to automatically rewrite imports, easing the migration.
+
+Security Implications
+=====================
+
+As with any new C code, especially code operating on potentially untrusted user
+input, there are risks of memory safety issues. The author plans on
+contributing integration with libfuzzer to enable fuzzing the ``_zstd`` code
+and ensure it is robust. Furthermore, there are a number of tests that exercise
+the compression and decompression routines. These tests pass without error when
+compiled with AddressSanitizer.
+
+Taking on a new dependency also always has security risks, but the ``zstd``
+library is mature, fuzzed on each commit, and `participates in Meta's bug bounty
+program <https://github.com/facebook/zstd/blob/dev/SECURITY.md>`_.
+
+How to Teach This
+=================
+
+Documentation for the new module is in the reference implementation branch. The
+documentation for other modules will be updated to discuss the deprecation of
+their existing import names, and how to migrate.
+
+Reference Implementation
+========================
+
+The `reference implementation <https://github.com/emmatyping/cpython/tree/zstd>`_
+contains the ``_zstd`` C code, the ``compression.zstd`` code, modifications to
+``tarfile``, ``shutil``, and ``zipfile``, and tests for each new API and
+integration added. It also contains the re-exports of other compression
+modules. Deprecations for the existing import names will be added once a
+decision is reached regarding the open issues.
+
+Rejected Ideas
+==============
+
+Name the module ``libzstd`` and do not make a new ``compression`` namespace
+---------------------------------------------------------------------------
+
+One option instead of making a new ``compression`` namespace would be to find
+a different name, such as ``libzstd``, as the import name. However, the issue
+of existing import names is likely to persist for future compression formats
+added to the standard library. LZ4, a common high speed compression format,
+has `a package on PyPI <https://pypi.org/project/lz4/>`_, ``lz4``, with the
+import name ``lz4``. Instead of solving this issue for each compression format,
+it is better to solve it once and for all by using the already-claimed
+``compression`` namespace.
+
+Open Issues
+===========
+
+Should we keep old compression imports?
+---------------------------------------
+
+It would be confusing to indefinitely have ``lzma`` and ``compression.lzma``
```

Comment:
Seems like the PEP has a definite answer here (above), should this open issue be removed?

Reply:
I wanted to lay out alternatives that were considered, but perhaps these open issues should go into rejected ideas.